libminifi/src/utils/StringUtils.cpp

```diff
@@ -46,8 +46,13 @@ std::string StringUtils::trim(const std::string& s) {
   return trimRight(trimLeft(s));
 }
 
-std::vector<std::string> StringUtils::split(const std::string &str, const std::string &delimiter) {
+template<typename Fun>
+std::vector<std::string> split_transformed(const std::string& str, const std::string& delimiter, Fun transformation) {
   std::vector<std::string> result;
+  if (delimiter.empty()) {
+    std::transform(str.begin(), str.end(), std::back_inserter(result), [] (const char c) { return std::string{c}; });
```

Comment:
This branch should apply `transformation` as well.

Reply:
Updated.